src/applications/proxy-rewrite/tests/redirects/index.unit.spec.js

```diff
@@ -29,3 +31,45 @@ describe('Redirect replaced pages', () => {
     expect(fakeWindow.location.href).to.be.undefined;
   });
 });
+
+describe('Validate crossDomainRedirects.json', () => {
+  // This string is technically the prod S3 bucket domain plus the path to this JavaScript bundle
+  // generated via Webpack so we can technically derive this from shared constants.
+  // However, it's hardcoded into TeamSite bundles as this URL, so hardcoding it instead adds that extra
+  // validation in case one of the constants changed one day.
+
+  const REDIRECT_ENTRY_SCRIPT =
+    'https://prod-va-gov-assets.s3-us-gov-west-1.amazonaws.com/generated/proxy-rewrite.entry.js';
+
+  const redirectsBySource = redirects.reduce((grouped, redirect) => {
+    const fullPath = `https://${redirect.domain}${redirect.src}`;
+    const items = grouped[fullPath] || [];
+    return {
+      ...grouped,
+      [fullPath]: items.concat(redirect.dest),
+    };
+  }, {});
+
+  Object.entries(redirectsBySource).forEach(([fullSource, destinations]) => {
+    it(`${fullSource} is a valid URL`, () => {
+      const url = new URL(fullSource);
+      expect(fullSource).to.be.equal(url.href);
+    });
+
+    it(`${fullSource} has a unique valid destination`, () => {
+      expect(destinations.length).to.be.equal(1);
+      const destUrl = new URL(destinations[0], 'https://www.va.gov');
+      expect(destinations[0]).to.be.equal(destUrl.pathname);
+    });
+
+    it(`${fullSource} contains the proxy-rewrite JavaScript bundle`, async () => {
+      const response = await fetch(fullSource);
+      const pageContents = await response.text();
+      const redirectEntryIncluded = pageContents.includes(
+        REDIRECT_ENTRY_SCRIPT,
+      );
+
+      expect(redirectEntryIncluded).to.be.true;
```

Comment:
I realize this doesn't fit the definition of an ideal unit test, since it's now dependent on external services (websites in this case) and is time-consuming. However, this would be super useful for our team. I'm wondering if we could check the branch name - if it contains `redirects` or `teamsite`, then run this test case, otherwise skip it.

@rianfowler what do you think?

Reply:
how are you going to convey to other people that this is a requirement to run this validation? We can't put comments in a json file so if someone updates those redirects who doesn't work with you or us, then this validation won't run unless they follow this convention.